### PR TITLE
Optional auth for execWebhook

### PR DESCRIPTION
**Pull request by Khaaz**
- Remove executing webhook with no webhook token but bot auth (which is no longer possible).
- Add an option to auth with your bot token when executing a webhook. It allows to use custom emojis that are not in the current server

### CODE REVIEW

File: lib/Client.js [hunk 1]
```diff
@@ -1929,13 +1930,14 @@ class Client extends EventEmitter {
     /**
     * Get a guild's data via the REST API. REST mode is required to use this endpoint.
     * @arg {String} guildID The ID of the guild
+    * @arg {Boolean} [withCounts=false] Whether the guild object will have approximateMemberCount and approximatePresenceCount
     * @returns {Promise<Guild>}
     */
-    getRESTGuild(guildID) {
+    getRESTGuild(guildID, withCounts = false) {
```
Comment (TTtie) [inline lib/Client.js]: Uh..  this is added in from #920 by accident, isn't it?

Comment (Khaaz) [reply lib/Client.js]: oh, I messed up somewhere thanks

File: lib/Client.js [hunk 2]
```diff
@@ -682,7 +683,7 @@ class Client extends EventEmitter {
     executeSlackWebhook(webhookID, token, options) {
         const wait = !!options.wait;
         options.wait = undefined;
-        return this.requestHandler.request("POST", (token ? Endpoints.WEBHOOK_TOKEN_SLACK(webhookID, token) : Endpoints.WEBHOOK_SLACK(webhookID)) + (wait ? "?wait=true" : ""), !token, options);
+        return this.requestHandler.request("POST", Endpoints.WEBHOOK_TOKEN_SLACK(webhookID, token) + (wait ? "?wait=true" : ""), false, options);
```
Comment (abalabahaha) [inline lib/Client.js]: Does this one not support auth?

Comment (Khaaz) [reply lib/Client.js]: To trigger this endpoint, you need to send slack-formatted data so I assumed auth was just not really necessary. We can probably add the option for auth anyway but I didn't think it was of any use

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**